### ISSUE TITLE
Training epochs 

### ISSUE THREAD

**Issue by rzcwade** [HawkAaron/RNN-Transducer#4]
Hi @HawkAaron ,

I have trained for over 100 epochs so far and the training loss is around -200000 and decreasing. Since I am not familiar with the RNNTLoss, the loss seemed quite large to me and I am not sure if it's converging. 

And also I ran the evaluation just to have a sneak peek at the PER, which is around 97% for a greedy search (beam search took way much longer).  

Did 200 epochs get you around 20% PER? I just want to know if my training is on the right track.

Thanks and your feedback would be really appreciated!

**Comment by HawkAaron** [comment]
Maybe you can check the inputs, here I just hard code into train.py, the phone set is 40 with blank, make sure the label is correct. And the training loss should be a positive number, and about 3 iterations, it would be less than 100. After 1 epoch, the training loss should be around 60, the cv loss should be around 40.

**Comment by rzcwade** [comment]
Thanks, @HawkAaron ! 
 It seems like the below phone set is what I have:
![image](https://user-images.githubusercontent.com/13773042/45891615-e3c81300-bd7a-11e8-876d-1f5d17521c20.png)
Is it because of my TIMIT dataset is different than your?

Thanks!

**Comment by HawkAaron** [comment]
This is the phone set I use, from kaldi recipe, 
{'<eps>': 0, 'sil': 1, 'aa': 2, 'ae': 3, 'ah': 4, 'aw': 5, 'ay': 6, 'b': 7, 'ch': 8, 'd': 9, 'dh': 10, 'dx': 11, 'eh': 12, 'er': 13, 'ey': 14, 'f': 15, 'g': 16, 'hh': 17, 'ih': 18, 'iy': 19, 'jh': 20, 'k': 21, 'l': 22, 'm': 23, 'n': 24, 'ng': 25, 'ow': 26, 'oy': 27, 'p': 28, 'r': 29, 's': 30, 'sh': 31, 't': 32, 'th': 33, 'uh': 34, 'uw': 35, 'v': 36, 'w': 37, 'y': 38, 'z': 39, '#0': 40, '#1': 41}

**Comment by rzcwade** [comment]
Thanks for sharing your phone set, @HawkAaron ! I could not just replace my phone.txt with yours since my Kaldi prep gave me 51 phone set corresponded training text as shown below:
![image](https://user-images.githubusercontent.com/13773042/45925231-234f4600-bec6-11e8-82a8-942568576f8a.png)

Maybe if you could share the version of TIMIT you used for training? 

Thanks very much!

**Comment by rzcwade** [comment]
The run.sh script you used is supposed to give you "mapping to 48 training phonemes". How did you end up using 40? 

Thanks!

**Comment by HawkAaron** [comment]
You can check `local/timit_data_prep.sh` in kaldi timit recipe, and change phone mapping to 39.
Actually, the 40 phone set includes `blank` for CTC training.

**Comment by rzcwade** [comment]
Now the loss is converging like you described! I need do more digging around the kaldi data prep :) It's strange that it works for 39-phone but not for 48. I really appreciate your help, @HawkAaron !

**Comment by chocolate9624** [comment]
@HawkAaron Do you reproduce the Transducer with 61 phonemes? 
In the code, the audio labels are all mapped to 48 classes. 
Thanks!

**Comment by HawkAaron** [comment]
@chocolate9624 It would be better to use 61 phonemes, I didn't try it.

**Comment by BuaaAlban** [comment]
> @chocolate9624 It would be better to use 61 phonemes, I didn't try it.

Hi，why is 61 phonemes better? And [here](https://github.com/HawkAaron/RNN-Transducer/issues/4#issuecomment-423714202) as you said, this is a 40 phonemes classes.
